pymmcore_plus/_tests/test_core.py

```diff
@@ -353,6 +353,28 @@ def test_set_property_events(core: CMMCorePlus):
     mock.assert_called_once_with("Camera", "AllowMultiROI", "1")
 
 
+def test_set_state_events(core: CMMCorePlus):
+    mock = MagicMock()
+    core.events.propertyChanged.connect(mock)
+    assert core.getState("Objective") == 1
+    core.setState("Objective", 3)
+    mock.assert_has_calls(
+        [
+            call("Objective", "State", "3"),
+            call("Objective", "Label", "Nikon 20X Plan Fluor ELWD"),
```

Comment:
Mentioning to expect both of these and only handle one of them would be great to include in a `working with properties and state devices` example.